Google.Api.Generator/Generation/ServiceYaml.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using SharpYaml;
+using SharpYaml.Serialization;
+
+namespace Google.Api.Generator.Generation
+{
+    public class ServiceYaml
+    {
+        public static Service ParseYaml(string yamlContentRaw)
+        {
+            var yamlContentTrim = yamlContentRaw.Trim();
+
+            // If the yaml starts with the `type` attribute, it should be discarded.

```

Comment:
I would clarify that only the type attribute should be discarded... I misread this to start with as "just ignore the whole YAML file".

Reply:
(It's not clear to me why we need to discard it, to be honest... shouldn't we validate that it has the expected value?)